bindings/java/src/org/sleuthkit/datamodel/CaseDatabaseFactory.java

```diff
@@ -436,25 +436,26 @@ private void createAccountTables(Statement stmt) throws SQLException {
 		
 		// References tsk_hosts
 		stmt.execute("CREATE TABLE tsk_os_account_realms (id " + dbQueryHelper.getPrimaryKey() + " PRIMARY KEY, "
-				+ "name TEXT NOT NULL, "	// realm name - host name or domain name
+				+ "realm_name TEXT DEFAULT NULL, "	// realm name - for a domain realm, may be null
 				+ "realm_addr TEXT DEFAULT NULL, "		// a sid/uid or some some other identifier, may be null
-				+ "host_id " + dbQueryHelper.getBigIntType() + " DEFAULT NULL, " // if the realm just comprises of a single host
-				+ "name_type INTEGER, "	// indicates if the realm name was  was expressed or inferred 
-				+ "UNIQUE(name, host_id), "
-				+ "FOREIGN KEY(host_id) REFERENCES tsk_hosts(id) )");
+				+ "realm_signature TEXT NOT NULL, "	// is either the realm_addr or the realm_addr, one of them must be known.  This is used to prevent duplicates.
+				+ "scope_host_id " + dbQueryHelper.getBigIntType() + " DEFAULT NULL, " // if the realm scope is a single host
```

Comment:
I think this comment should say that the signature can be either the realm name or the realm address.